Added d3-jetpack dependency

beemyfriend opened the pull request:
I'm a big fan of d3-jetpack. I use it in d3 the same way I use magrittr in R. I think it'd be nice to add a 
`d3_dep_jetpack` function to make writing d3 functions in R strings easier/faster.

The following should now work:

```
library(htmltools)
library(d3r)

t <- tagList(tags$script(HTML(sprintf(
  "
    var x = 5;

    var svg = d3.select('body')
      .append('svg');

    svg.append('rect')
      .at({
        x: 100,
        y: 100,
        width: 100,
        height: 100
      })
      .st({
        fill: 'blue',
        stroke: 'purple'
      });

    svg.append('circle')
      .at({
        cx: 250,
        cy: 250,
        r: 50
      })
      .st({
        fill: 'purple',
        stroke: 'blue'
      })
  "
))), d3_dep_jetpack())
browsable(t)
```
The following should also work:

```
t <- tagList(tags$script(HTML(sprintf(
  "
  var svg = d3.select('body')
  .append('svg');

  test_data = [{x: 250, y: 250}, {x: 300, y: 300}, {x: 300, y: 100}];
  
  svg.appendMany(test_data, 'circle')
    .at({
      cx: function(d){return d.x},
      cy: function(d){return d.y},
      r: 50
    })
    .st({
      fill: 'purple',
      stroke: 'blue'
    });
  "
))), d3_dep_jetpack())
browsable(t)
```

timelyportfolio commented:
@beemyfriend, very happy and appreciative that you spent the time to submit this pull.  I have up to now avoided adding non-core `d3` dependencies, but I think `d3-jetpack` is an exception.  A couple of things:

1. We should give attribution to `d3-jetpack` authors in `DESCRIPTION`.  I am happy to do this after acceptance, or feel free to make the change.

2.  I like to add test, so that I don't miss an update.  Here is where I test for most recent `d3` [test_deps.R](https://github.com/timelyportfolio/d3r/blob/master/tests/testthat/test_deps.R).

3. I think an example would be very nice for roxygen even if it does not really show `d3-jetpack` functionality.  I can see a user not knowing that they still need to add/append `d3_dep_v4()` in the `tagList`.

4. I would like to add you as contribution in `DESCRIPTION`.  Feel free to modify pull or I will after acceptance.

5  We should minor increment `d3r` version, update date, and add NEWS item.  Again, happy to do, or if you want to complete the learning experience, have at it.


**Thanks again!!!**

beemyfriend commented:
Awesome! I definitely need to learn so I'll work on those bullet points and will hopefully get them done by Thursday. Also, I think `d3r` comes with `selection-multi.` I haven't used `ndjson` at all, but a command line interface sounds perfect for an `R` environment.

timelyportfolio commented:
Great, let me know if you need any help and absolutely no rush.  Also, if you would like to see how the CRAN submission works once we are all done, I'll be glad to include you in that.

beemyfriend commented:
I came to the realization that unpkg version is just the d3-jetpack module, while the package I used in `d3_dep_jetpack()`. I'll make the change to just the d3-jetpack module, but I need to understand html dependencies before I do that. I should get it done by tonight.

timelyportfolio commented:
@beemyfriend one tick with unpkg is append `/` and you can see if a build is included.  In this case we are lucky with https://unpkg.com/d3-jetpack@2.0.7/build/d3-jetpack.js.

beemyfriend commented:
I didn't add the final test that tries to match the local version number to the github version number because:

1. - I can't seem to find the `github` package.
2. - The version number found on 'https://api.github.com/repos/gka/d3-jetpack/releases/latest' does not match the version found on  https://unpkg.com/d3-jetpack@2.0.7. (1.0.6 vs 2.0.7)

I added all the other tests and they work fine. Also, I'm still using the the full d3v4+jetpack source because I still don't understand how to use modules with `htmltools`. Using the `d3-jetpack.js` module would obviously be better because it is only 14.5 kb as opposed to 471.4 kb for `d3v4+jetpack.js`. Most of `d3v4+jetpack` is just `d3v4`. I will get rid of `d3v4+jetpack` as soon as I understand how to build `d3_dep_jetpack()` ontop of `d3_dep_v4()`.

beemyfriend commented:
ok, I updated the code so that it is only using the `d3-jetpack` module instead of the full `d3v4+jetpack` package. The packages should be much lighter now. I also updated the documentation.

The only thing I couldn't do was the version-comparison test for reasons explained above. 

Everything should be working fine. If you decide to put this on CRAN, then I would definitely love to see how the submission process works.

Also, I don't really know what's going on with the travis-cl build. It keeps on failing, what should I do to not fail travis-cl?

timelyportfolio commented on DESCRIPTION:
```diff
@@ -14,6 +14,16 @@ Authors@R: c(
+      person(
+        "Adam", "Pearce"
+        , role = c("aut")
+        , comment = "core contributor to d3-jetpack"
```
@beemyfriend, where are you as `CTB`?

timelyportfolio commented:
@beemyfriend, well done and thanks so much.  Just left one more comment to add you as ` CTB`.  I hope this well be your first contributor role in a CRAN package.

timelyportfolio commented:
@beemyfriend, just submitted issue https://github.com/gka/d3-jetpack/issues/37 requesting a Github tagged release for the newest version.  For now, I will probably add a test checking unpkg.com https://unpkg.com/d3-jetpack/package.json as another way to verify that `d3r` is current.

beemyfriend commented:
Cool, I'm excited to see how you create the test.

timelyportfolio commented:
Well, pull accepted.  Thanks for working through all the steps with me and your eagerness to learn.  I hope it was a pleasant experience.  I will test thoroughly on my side and submit to CRAN by the end of the week.  Please let me know if you have any interest in seeing how that process works.
